File: SupplierPortal.Web/Modules/ServerTypes/Market/OfferSupplierColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { OfferSupplierRow } from "./OfferSupplierRow";

export interface OfferSupplierColumns {
    Id: Column<OfferSupplierRow>;
    CompanyName: Column<OfferSupplierRow>;
    SupplierId: Column<OfferSupplierRow>;
    OfferId: Column<OfferSupplierRow>;
}

export class OfferSupplierColumns extends ColumnsBase<OfferSupplierRow> {
    static readonly columnsKey = 'Market.OfferSupplier';
    static readonly Fields = fieldsProxy<OfferSupplierColumns>();
}

File: SupplierPortal.Web/Modules/Market/OfferSupplier/OfferSupplierColumns.cs
using MovieTutorial.MovieDB;
using Serenity.ComponentModel;
using System.ComponentModel;

namespace SupplierPortal.Market.Columns;

[ColumnsScript("Market.OfferSupplier")]
[BasedOnRow(typeof(OfferSupplierRow), CheckNames = true)]
public class OfferSupplierColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    [Visible(false)]
    public int Id { get; set; }
    [Visible(false)]
    public int SupplierId { get; set; }
    [EditLink]
    public string CompanyName { get; set; }
    public string Email { get; set; }
    public string OfficalName { get; set; }
    public string OfficalSurname { get; set; }
    public string CallNumber { get; set; }
    [Visible(false)]
    public int OfferId { get; set; }
}